### PR TITLE
Multiple common names & including taxonomic rank on the species card

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

None

## Description of Changes

- Replace all references to commonName with commonNames
- Change commonName type from a string to an array of strings
- Include taxonomic rank in the Species Card to indicate whether taxon is a species, genus, family, etc.
- Include kingdom and rank in biohub's taxonomy API response

## Testing Notes
- *Must be run locally with this BioHub PR: https://github.com/bcgov/biohubbc-platform/pull/237*
- Species can have multiple common names
- Species card should always include the rank label

Will need to be merged with https://github.com/bcgov/biohubbc-platform/pull/237

### CODE REVIEW

File: database/src/migrations/release.0.34/api_get_eml_data_package.sql [hunk 1]
```diff
@@ -294,7 +294,7 @@ begin
         <taxonomicClassification>
 				  <taxonRankName>%1$s</taxonRankName>
 				  <taxonRankValue>%2$s</taxonRankValue>
-				  <commonName>%3$s</commonName>
+				  <commonNames>%3$s</commonNames>
```
Comment (NickPhura) [inline database/src/migrations/release.0.34/api_get_eml_data_package.sql]: Looks like this just got caught in the find/replace.

Since this is an old migration, we should undo this change.

Comment (mauberti-bc) [reply database/src/migrations/release.0.34/api_get_eml_data_package.sql]: Oops, I'll make sure to exclude migrations when find-replacing. Thanks

File: app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx [hunk 2]
```diff
@@ -190,8 +196,24 @@ const AsyncAutocompleteDataGridEditCell = <DataGridType extends GridValidRowMode
       )}
       renderOption={(renderProps, renderOption) => {
         return (
-          <Box component="li" {...renderProps}>
-            {renderOption.label}
+          <Box
```
Comment (al-rosenthal) [inline app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx:205]: this should probably be swapped out for: `<List>` and `<ListItem>` components

File: app/src/components/species/components/SpeciesCard.tsx [hunk 3]
```diff
@@ -1,45 +1,93 @@
...
-  commonName: string | null;
+  commonNames: string[];
   scientificName: string;
   tsn: number;
+  rank: string;
+  kingdom: string;
 }
 
 const SpeciesCard = (props: ISpeciesCard) => {
+  const rankColors = [
```
Comment (al-rosenthal) [inline app/src/components/species/components/SpeciesCard.tsx]: nit: this could probably be placed in the constants folder and imported to be used here. It makes it easier to use in other places in code while only have a single source of truth to maintain

**Review by al-rosenthal — COMMENTED** [review]
I tried running this locally and ran into issues viewing the survey page. Looks like an issue with the open api spec on the view endpoint. It is missing the `commonNames` property

**Comment by mauberti-bc** [comment]
> I tried running this locally and ran into issues viewing the survey page. Looks like an issue with the open api spec on the view endpoint. It is missing the `commonNames` property

I should have mentioned that this PR needs to be tested with this BioHub branch: https://github.com/bcgov/biohubbc-platform/pull/237. That fixes the issue for me.

**Review by NickPhura — APPROVED** [review]
